data/random-teams.ts

```diff
@@ -1308,19 +1307,25 @@ export class RandomTeams {
 				!counter.technician ||
 				hasMove['tailslap'] ||
 				hasAbility['Punk Rock'] ||
+				// For Doubles Alolan Persian
 				movePool.includes('snarl')
 			);
 		case 'Tinted Lens':
 			return (
+				// For Sigilyph
 				hasMove['defog'] ||
-				(species.id === 'butterfree') ||
+				// For Butterfree
+				(hasMove['hurricane'] && hasAbility['Compound Eyes']) ||
 				(counter.Status > 2 && !counter.setupType)
 			);
 		case 'Torrent':
+			// For Inteleon-Gmax and Primarina
 			return (hasMove['focusenergy'] || hasMove['hypervoice']);
```

Comment:
I'd rather cull based on the abilities (Sniper and Liquid Voice) that are preferred to Torrent.
```suggestion
			return hasAbility['Sniper'] || hasAbility['Liquid Voice'];
```

Reply:
that wouldn't work. Primarina only wants Liquid Voice in doubles, and even then only when it has Hyper Voice. Hyper Voice is the proper cull here.

Same goes for Sniper; Regular inteleons without Focus Energy don't want to have Torrent rejected. Only the ones *with* focus energy want sniper. The rest want Torrent.

It's a simple matter of how hasAbility only checks for *available* abililties and not anything else

Reply:
Oh, do we prefer Torrent + Hydro Pump in singles? this is fine, then, I suppose.